topology/topology_parameter.h

```diff
@@ -458,6 +461,52 @@ class Gaussian2DParameter : public TopologyParameter
 
 
 /**
+ * Gamma parameter p(d) = d^(kappa-1)*exp(-d/theta)/(theta^kappa*gamma(kappa))
+ */
+class GammaParameter : public RadialParameter
+{
+public:
+  /**
+   * Parameters:
+   * kappa    - shape
+   * theta    - scale
+   */
+  GammaParameter( const DictionaryDatum& d )
+    : RadialParameter( d )
+    , kappa_( 1.0 )
+    , theta_( 1.0 )
+  {
+    updateValue< double >( d, names::kappa, kappa_ );
+    updateValue< double >( d, names::theta, theta_ );
+    if ( kappa_ <= 0 )
+      throw BadProperty(
+        "topology::GammaParameter: "
+        "kappa > 0 required." );
+    if ( theta_ <= 0 )
+      throw BadProperty(
+        "topology::GammaParameter: "
+        "theta > 0 required." );
+  }
+
+  double
+  raw_value( double x ) const
+  {
+    return std::pow( x, kappa_ - 1 ) * std::exp( -x / theta_ )
+     / ( std::pow( theta_, kappa_ ) * gsl_sf_gamma( kappa_ ) );
```

Comment:
For efficiency, `1 / ( std::pow( theta_, kappa_ ) * gsl_sf_gamma( kappa_ ) )` should be pre-computed in the constructor so that one only needs a single multiplication here. Also, precompute `beta_ = 1.0 / theta_` in the constructor, so that you avoid the division in the `exp()` argument.